Azure/azure-cli
# [RDBMS] Add MigrationRuntimeResourceId to properties of creating migration request in CLI

**Related command**
`az postgres flexible-server migration create --subscription sub --resource-group testGroup --name testserver --migration-name testmigration --properties "migrationConfig.json"`

**Description**
Introducing new optional property MigrationRuntimeResourceId that will be responsible for migrating data between source and target server in support of private endpoint migration. This property is to be defined in JSON file passed in properties argument for creating a migration

**Testing Guide**
Manually
Help:
![image](https://github.com/Azure/azure-cli/assets/69922333/af6aa8ac-b1bd-4a24-8e99-05c6807bfc66)

Pass Test:
![image](https://github.com/Azure/azure-cli/assets/69922333/6f2398e3-7e6d-48f1-b35b-ebd05dafb1da)
<img width="810" alt="image" src="https://github.com/Azure/azure-cli/assets/69922333/5ce2f2d2-94b3-4890-b740-821f5b71cc69">

Fail Test:
'postgres' 'flexible-server' 'migration' 'create' '--subscription' 'ac0271d6-426b-4b0d-b88d-0d0e4bd693ae' '--name' 'cli-test-server-pg-flex-eastus2euapac0271d6426b4b0db8' '--migration-name' 'nasctestmigration1' '-g' 'cli-test-do-not-delete-rg-eastus2euapac0271d6426b4b0d' '--properties' 'C:\Users\nasc\migration_test.json' '--migrationRuntimeResourceId' '/subscriptions/ac0271d6-426b-4b0d-b88d-0d0e4bd693ae/resourceGroups/temp-ss-postgresql-240102-0611-760428cd/providers/Microsoft.DBforPostgreSQL/servers/testmigration-nasc1' 
WARNING: Checking the existence of the resource group 'cli-test-do-not-delete-rg-eastus2euapac0271d6426b4b0d'...
WARNING: Resource group 'cli-test-do-not-delete-rg-eastus2euapac0271d6426b4b0d' exists ? : True 
**ERROR: Migration runtime resource ID provided should be Flexible server.**

'postgres' 'flexible-server' 'migration' 'create' '--subscription' '5c5037e5-d3f1-4e7b-b3a9-f6bf94902b30' '--resource-group' 'Aditya_PM' '--name' 'test-flexible-endpoint' '--migration-name' 'nasctestmigration3' '--properties' 'C:\Users\nasc\OneDrive - Microsoft\Documents\migration_test.json' '--migrationRuntimeResourceId' '/subscriptions/5c5037e5-d3f1-4e7b-b3a9-f6bf94902b30/resourceGroups/Aditya_PM/providers/Microsoft.DBforPostgreSQL/flexibleServers/test-flexible-endpoint' 
WARNING: Checking the existence of the resource group 'Aditya_PM'...
WARNING: Resource group 'Aditya_PM' exists ? : True 
**ERROR: Migration Runtime server is same as Target Flexible server. Please change the values accordingly.**


**History Notes**
[RDBMS] `az postgres flexible-server migration create`: Add private endpoint support for migrations by providing migration runtime resource ID

---

This checklist is used to make sure that common guidelines for a pull request are followed.
